TenElephants/TenElephants/Controllers/IngredientsPageController.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  IngredientsPageController.swift
+//  TenElephants
+//
+//  Created by Полина Скалкина on 16.12.2021.
+//
+
+import Foundation
+import UIKit
+
+final class IngredientsPageController: UIViewController {
+    private enum Constants {
+        static let cellId = "IngredientTableCell"
+        static let ingredientsHeaderCellId = "IngredientsHeaderTableCell"
+        static let ingredientCellHeight = 60.0
+        static let headerCellHeight = 70.0
+    }
+    
+    private lazy var tableView = UITableView(frame: .zero)
+    
+    private let dataProvider: IngredientsDataProvider
+    private var ingredientsData: IngredientsUIData?
+    
+    init(
+        dataProvider: IngredientsDataProvider
+    ) {
+        self.dataProvider = dataProvider
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("Init from coder is not available")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        fetchData()
+        configTableView()
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        tableView.frame = view.bounds
+    }
+    
+    private func fetchData() {
+        dataProvider.fetchIngredientsList { [weak self] result in
+            switch result {
+            case let .failure(error):
+                print(error.localizedDescription) // Пока не решили, что придумаем для ошибок
+            case let .success(data):
+                self?.ingredientsData = data
+                DispatchQueue.main.async { [weak self] in
+                    self?.tableView.reloadData()
+                }
+            }
+        }
+    }
+}
+
+
+extension IngredientsPageController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0: return 1
+        case 1: return ingredientsData?.ingredients.count ?? 0
+        default: return 0
+        }
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        2
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        switch indexPath.section {
+        case 0:
```

Comment:
nit: enum of section types?
`
enum Section {
   case header
   case scrollable
}
`